tasks/test_results_processor.py

```diff
@@ -149,125 +195,149 @@ def _bulk_write_tests_to_db(
         commitid: str,
         upload_id: int,
         branch: str,
-        parsed_testruns: List[Testrun],
+        parsing_results: List[ParsingInfo],
+        network: list[str] | None,
         flaky_test_set: set[str],
         flags: list[str],
     ):
-        test_data = []
+        test_data = {}
         test_instance_data = []
         test_flag_bridge_data = []
-        daily_totals = dict()
+        daily_totals: dict[str, dict[str, str | int | list[str]]] = dict()
```

Comment:
Not something to do in this PR, but I would consider creating a custom object/type/dataclass to simplify this complex type annotation.